packages/cli/src/commands/deploy/index.ts

```diff
@@ -153,6 +153,27 @@ export default async (client: Client) => {
         return 1;
       }
     }
+
+    // Validate localConfig.projectSettings
+    if (localConfig.projectSettings) {
+      const allowedProperties = [
+        'buildCommand',
+        'devCommand',
+        'installCommand',
+        'framework',
+        'name',
+        'nodeVersion',
+        'commandForIgnoringBuildStep',
+        'rootDirectory',
+        'sourceFilesOutsideRootDirectory',
+      ];
+      for (const key of Object.keys(localConfig.projectSettings)) {
+        if (!allowedProperties.includes(key)) {
+          output.error(`Disallowed property ${key} found in "projectSettings"`);
```

Comment:
```suggestion
          output.error(`Unknown property ${key} found in "projectSettings"`);
```

We should consider this unknown just like if there was any other unknown property.

What happens when you do `{ projectSettings: { "foo": "bar" } }`?

Reply:
I think you'll eventually hit an API error when this object is validated by the API using the schema

Reply:
Lets remove this and validate in the API instead of the CLI

Reply:
So the create-deployment api route will need to use a different schema than update-project. Okay ✅